src/journal/journald-syslog.c

```diff
@@ -441,17 +454,21 @@ void server_process_syslog_message(
         server_dispatch_message(s, iovec, n, m, context, tv, priority, 0);
 }
 
-int server_open_syslog_socket(Server *s) {
-
-        static const union sockaddr_union sa = {
-                .un.sun_family = AF_UNIX,
-                .un.sun_path = "/run/systemd/journal/dev-log",
-        };
+int server_open_syslog_socket(Server *s, const char *syslog_socket) {
         int r;
 
         assert(s);
+        assert(syslog_socket);
 
         if (s->syslog_fd < 0) {
+                union sockaddr_union sa = {
```

Comment:
This change pattern seems almost identical in three files, is it worth collapsing this pattern into some function?

Reply:
hmm, we need to allocate the union on the stack. Stack allocation is not really something we can sanely do. And the initialization is pretty trivial... not sure it's necessary to do anything about it...